digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -335,19 +336,35 @@ pub async fn get_related_for_assets(
             .add(asset_grouping::Column::Verified.is_null())
     };
 
-    let grouping = asset_grouping::Entity::find()
+    let grouping = if options.show_collection_metadata {
+        asset_grouping::Entity::find()
+        .find_also_related(asset_data::Entity)
         .filter(asset_grouping::Column::AssetId.is_in(ids.clone()))
         .filter(asset_grouping::Column::GroupValue.is_not_null())
         .filter(cond)
         .order_by_asc(asset_grouping::Column::AssetId)
         .all(conn)
-        .await?;
-    for g in grouping.into_iter() {
+        .await?
+    } else {
+        asset_grouping::Entity::find()
+        .filter(asset_grouping::Column::AssetId.is_in(ids.clone()))
+        .filter(asset_grouping::Column::GroupValue.is_not_null())
+        .filter(cond)
+        .order_by_asc(asset_grouping::Column::AssetId)
+        .all(conn)
+        .await?
+        .into_iter()
+        .map(|g| (g, None))
+        .collect::<Vec<_>>()
+    };
```

Comment:
There is some duplication here and the extra mapping to inject None. Its worth another pass to see if this can be cleaned up.